tools/pylint-extensions/azure-pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2865,10 +2865,79 @@ def visit_import(self, node):
 
 
 # [Pylint] custom linter check for invalid use of @overload #3229
-# [Pylint] Custom Linter check for Exception Logging #3227
+
+
+class DoNotLogExceptions(BaseChecker):
+
+    """Rule to check that exceptions aren't logged"""
+
+    name = "do-not-log-exceptions"
+    priority = -1
+    msgs = {"C4766": (
+            "Do not log exceptions. See Details:"
+            " https://azure.github.io/azure-sdk/python_implementation.html#python-logging-sensitive-info",
+            "do-not-log-exceptions",
+            "Do not log exceptions in levels other than debug, it can otherwise reveal sensitive information",
+            ),
+            }
+
+    def visit_try(self, node):
+        """Check that exceptions aren't logged in exception blocks.
+           Go through exception block and branches and ensure error hasn't been logged.
+        """
+        # Return a list of exception blocks
+        except_block = node.handlers
+        # Iterate through each exception block
+        for nod in except_block:
+            # Get exception blocks with an exception name
+            if nod.name is not None:
+                exception_name = nod.name.name
+                self.check_for_logging(nod.body, exception_name)
+
+    def check_for_logging(self, node, exception_name):
+        """ Helper function - checks nodes to see if logging has occurred at all
+            levels.
+        """
+        levels_matches = [".warning", ".error", ".info", ".debug"]
+        for j in node:
+            if isinstance(j, astroid.Expr):
+                expression = j.as_string().lower()
+                if any(x in expression for x in levels_matches) and "logger" in expression:
+                    # Check for variables after strings
+                    end_finder = expression.rfind("'")
```

Comment:
This was added to help fix a false positive from testing against the SDKs. `end_finder` finds the end of any strings being logged which should give access to just the variables after. For the code example below, this should mean that the checker is triggered for the variable 'exception' rather than the usage of the word 'exception' in the string.
`except Exception as exception:`
`    logger.warning("There has been an exception", exception)`
If there is only a string and no variables after, it should look at the string instead.
As for the single quote, when the checker looks at the string it seems to convert double quotes to single quotes. Think the culprit might be `expression = j.as_string()`. So while the above code uses double quotes, when my checker runs they switch to single. Happy to investigate this further if you'd like.

Reply:
ah gotcha!